src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,163 @@
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                shoppingCart = getCartFromResultSet(resultSet);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get product by " + id + " !", ex);
+        }
+        shoppingCart.setProducts(getProductsFromCart(shoppingCart.getId()));
+        return Optional.of(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart cart) {
+        String queryToDeleteProducts = "DELETE FROM shopping_carts_products WHERE cart_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(queryToDeleteProducts)) {
+            statement.setLong(1, cart.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete products from shopping cart - "
+                    + cart, e);
+        }
+        addProductsToCart(cart);
+        return cart;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM shopping_carts WHERE cart_id=?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            int updates = statement.executeUpdate();
+            return updates > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete cart " + id, e);
+        }
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        List<ShoppingCart> shoppingCarts = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement("SELECT * FROM shopping_carts");
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                ShoppingCart shoppingCart = getCartFromResultSet(resultSet);
+                shoppingCarts.add(shoppingCart);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Get all carts is failed!", e);
+        }
+        return shoppingCarts;
+    }
+
+    private ShoppingCart getCartFromResultSet(ResultSet resultSet) throws SQLException {
+        Long cartId = resultSet.getLong("cart_id");
+        Long userId = resultSet.getLong("user_id");
+        return new ShoppingCart(cartId, userId);
+    }
+
+    private List<Product> getProductsFromCart(Long cartId) {
+        String query = "SELECT * FROM shopping_carts_products "
+                + "INNER JOIN products "
+                + "ON shopping_carts_products.product_id = products.product_id "
+                + "WHERE shopping_carts_products.cart_id = ?";
```

Comment:
такі запити можна спрощувати:
```suggestion
        String query = "SELECT * FROM shopping_carts_products cp "
                + "INNER JOIN products p"
                + "ON cp.product_id = p.product_id "
                + "WHERE cp.cart_id = ?";
```

Reply:
ok. next time will use